Return constexpr char* from name function in OptimizationPass

rdspring1 opened the pull request:
The name for the derived child class is destroyed before calling `Trace::endEvent`, so a bad name is created and chrome tracing cannot open the trace. This PR changes the return type of `OptimizationPass::name` from `std::string` to `constexpr char*`.

Error:
```
While importing:
SyntaxError: Bad control character in string literal in JSON at position 307792 (line 3203 column 14)
    at JSON.parse (<anonymous>)
    at new TraceEventImporter (chrome://tracing/tracing.js:6244:19)
    at Import.createImporter_ (chrome://tracing/tracing.js:2033:8)
    at chrome://tracing/tracing.js:2025:167
    at Task.run (chrome://tracing/tracing.js:3149:95)
    at runAnother (chrome://tracing/tracing.js:3152:371)
    at runTask (chrome://tracing/tracing.js:2913:57)
    at processIdleWork (chrome://tracing/tracing.js:2918:116)
    at window.requestIdleCallback.timeout (chrome://tracing/tracing.js:2911:81)
```

rdspring1 commented:
!test

rdspring1 commented:
!test

wujingyue commented on csrc/instrumentation.h:
```diff
@@ -81,15 +81,15 @@ class Trace : public NonCopyable {
 
  private:
-  const char* event_name_ = nullptr;
+  std::string event_name_ = nullptr;
```
```suggestion
  const std::string event_name_;
```

wujingyue commented:
Great -- thanks for figuring this out!

rdspring1 commented:
@wujingyue I changed the return type of `OptimizationPass::name` from `std::string` to `constexpr char*`. I reverted the changes to `TraceScope` in `instrumentation.h`.

**Why?** Using `constexpr` to minimize latency of instrumentation.

wujingyue commented on csrc/preseg_passes/mark_aliases_prepare.h:
```diff
@@ -19,7 +19,7 @@ class MarkAliasesPreparePass : public OptimizationPass<MarkAliasesPreparePass> {
  protected:
   static void runPass(Fusion* fusion);
-  static std::string name() {
+  static constexpr const char* name() {
```
Nit: Consider `constexpr std::string_view` to be more type safe. There are too many ways to mess up a raw char pointer.

rdspring1 commented:
!test
